src/main/java/bg/libapp/libraryapp/service/UserService.java

```diff
@@ -0,0 +1,91 @@
+package bg.libapp.libraryapp.service;
+
+import bg.libapp.libraryapp.exceptions.UserNotFoundException;
+import bg.libapp.libraryapp.model.dto.user.UserEditDTO;
+import bg.libapp.libraryapp.model.dto.user.UserPasswordChangeDTO;
+import bg.libapp.libraryapp.model.dto.user.UserRoleChangeDTO;
+import bg.libapp.libraryapp.model.dto.user.UserViewDTO;
+import bg.libapp.libraryapp.model.entity.User;
+import bg.libapp.libraryapp.model.mappers.UserMapper;
+import bg.libapp.libraryapp.repository.UserRepository;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class UserService {
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    public UserService(UserRepository userRepository, PasswordEncoder passwordEncoder) {
```

Comment:
Is it on purpose that you don't have @autowire annotation?

Reply:
I think I have just forgotten to add it.
I'll fix it.